File: fundamentals/prime/prime.ts
export {}

function isPrime(num: number): boolean {
    if( num <= 1 ) return false;

    for(let i =2; i*i <= num; i++){
        if(num % i === 0) return false;
    }
    return true;
}

const number: number = 29;
console.log(isPrime(number));

File: fundamentals/prime/prime.py
import math

def is_prime(n):
    if n <= 1:
        return False
    i = 2
    while i * i <= n:
        if n % i == 0:
            return False
        i += 1
    return True

def is_prime_math(n):
    if n <= 1:
        return False
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return False
    return True

num = 25
if is_prime(num):
    print(num, "is a prime number.")
else:
    print(num, "is not a prime number.")
    
    
if is_prime_math(num):
    print(num, "is a prime number (using math method).")
else:
    print(num, "is not a prime number (using math method).")    